src/movies.js

```diff
@@ -1,16 +1,97 @@
 // Iteration 1: All directors? - Get the array of all directors.
+function getAllDirectors (movies){
+    let mapped = movies.map(function(movie){
+        return movie.director;
+    });
+    return mapped; 
+}
+
+//console.log(getAllDirectors(movies))
+
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors. How could you "clean" a bit this array and make it unified (without duplicates)?
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
+function howManyMovies(movies){
+    const dramaMovies = movies.filter(function(dramaMovie){
+        if (dramaMovie.director === "Steven Spielberg" && dramaMovie.genre.includes("Drama")){
+            return true;
+        }
+    }) 
+    return dramaMovies.length
+}
+
+//console.log(howManyMovies(movies));
+
 
 // Iteration 3: All rates average - Get the average of all rates with 2 decimals
 
+function ratesAverage(movies){
+      const filterRate = movies.map(function(movie){
+        return movie.rate;
+      });
+      //return filterRate;
+      const sumRate = filterRate.reduce(function(acc , val){
+        return acc + val;
+      }, 0);  
+      return Math.round((sumRate/movies.length)*100)/100;
+    };
+  
+//console.log(ratesAverage(movies));
+
+
+
+
 // Iteration 4: Drama movies - Get the average of Drama Movies
 
-// Iteration 5: Ordering by year - Order by year, ascending (in growing order)
+function dramaMoviesRate(movies){
+    if (movies.length === 0){
+    return 0;
+    } else {
+        const filterDrama = movies.filter(function(movieDrama){
+            return movieDrama.genre.includes("Drama");
+        });
+
+        const ratesDrama = filterDrama.reduce(function(acc, val){
+            return acc + val.rate;
+        },0)
+        
+        const averageDrama = ratesDrama / filterDrama.length;
+        return Math.round((averageDrama)*100)/100;
+}
+};
+
+// //console.log(dramaMoviesRate(movies))
+
+
+
+
+// // Iteration 5: Ordering by year - Order by year, ascending (in growing order)
+
+function orderByYear(movies){
+  const mapped = movies.map(function(movie){
+    return [movie.title ,movie.year]
+  });
+  const sorted = mapped.sort(function(a,b){
+    return b-a
+  });
+  return sorted;
+}
```

Comment:
The array `mapped` here will be an array of arrays. Each element will be an array of two elements title and year. This will get complex.

A simpler way is to just use `sort()`. In order to not mutate the original array, use `slice()` to make a copy of the array before sorting.

```
function orderByYear(array) {
  const sorted = array.slice().sort((a, b) => {
    if (a.year !== b.year) {
      return a.year - b.year;
    } else {
      return a.title.localeCompare(b.title);
    }
  });
  return sorted;
}
```

Remember you need to sort alphabetically in case the years are equal. `localeCompare` can do that for you.